woocommerce/woocommerce-gutenberg-products-block
# Price Filter: Introduce view block

## What

- This PR closes two issue as the same time (#11306 and woocommerce/woocommerce-blocks#10813) because the Price Filter block was used for experimenting new block structure.
- This PR introduces a new term called view block. This isn't a Gutenberg or WooCommerce term. With the block structure introduced in this PR, filter blocks don't need to fetch and process the data from Store API themselves, instead they just need to consume the data passed through context, that's why I called them view blocks. See https://github.com/woocommerce/woocommerce/issues/42178 for more detail.
- Move the logic calculating and enqueuing Interactivity Store to the Collection Filters block.
- The store data is passed to inner filter blocks through a context named `filterData`.
- Inner filter blocks (the Collection Price Filter in this case) consume that context to build the view. 
- View blocks must declare an attribute named `filterType` with a default value set to one of these strings to indicate their type: `price`, `attribute` , `stock` , or `rating`. Collection Filters block uses this attribute to build the filter data.
- View blocks can have their own data:
    - State: `rangeStyle`  in this case.
    - Attributes: `showInputFields`, `inlineInput`.

Fixes woocommerce/woocommerce#42178 woocommerce/woocommerce-blocks#10813 

## Why

By delegating the data fetching and processing to the Collection Filters block, we

- simplify the filter block code and logic.
- reducing the duplication between filter blocks.

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Add Product Collection > Collection Filters > Collection Price Filters to a new page.
2. See the block work in the editor preview and on the front end.

  


## WooCommerce Visibility

Required:


## Checklist

Required:


Conditional:


## Changelog  

> N/a
